### PR TITLE
Add dispute steps to resolve to the transaction details page.

### LINKED ISSUE #6925: Transaction Details > Dispute Details > Steps to resolve for disputes
Screenshot 2023-08-04 at 09 20 12 copy 2

> **Note**
> **Review the claim** is not part of this issue – now handled in issue #7156.

> **Note**
> Steps to resolve for **Inquiries** is not part of this issue – now handled in issue #7245.

**Link targets**
* `Email the customer` – a mailto link with the following prefilled detail:
	* Recipient: customer email
	* Subject: `Problem with your purchase from {store name} on {order date: YYYY-MM-DD}?
	* Body:
		```
		Hello {customer name}

		We noticed that on {dispute date: YYYY-MM-DD}, you disputed a {disputed amount with explicit formatting ($123 USD)} from {order date: YYYY-MM-DD}. We wanted to contact you to make sure everything was all right with your purchase and see if there's anything else we can do to resolve any problems you might have had.

		Alternatively, if the dispute was a mistake, you could easily withdraw it by calling the number on the back of your card. Thank you so much – we appreciate your business and look forward to working with you.
		```
	* `guidance on dispute withdrawal` – TBD (marked as "Doc in progress" in Figma – see pbUcTB-3oM-p2)

**Tooltips**
* Tooltips should use the `ClickTooltip` component.
* Challenge – `Challenge the dispute if you consider the claim invalid. You'll need to provide evidence to back your claim. Keep in mind that challenging doesn't ensure a resolution in your favor.`
* Accept – `Accepting this dispute will automatically close it. Your account will be charged a {fee amount, formatted explicitly: $15 USD} fee, and the disputed amount will be refunded to the cardholder.`]

#### Additional details
* See related issue for inquiries: #7245
* See figma: RiEQmaKRI7u9USAI3lyZz0VX-fi-6978_70329

### PR DESCRIPTION
**Pull request by shendy-a8c**
Fixes https://github.com/Automattic/woocommerce-payments/issues/6925.

> **Note**
> This PR is behind feature flag `_wcpay_feature_dispute_on_transaction_page`

#### Changes proposed in this Pull Request

Add the 'Steps to resolve' section on the transaction details page for disputes (**not inquiries** – handled in issue #7245).

steps

**Inquiries** will not see the steps rendered:
image

Notes:
- The step **Review the claim** is not part of this PR's scope – now handled in issue #7156.
- The text 'guidance on dispute withdrawal' from the third step is supposed to link to a doc ~but it's not yet ready.~ This has been published.
- 

#### Testing instructions

* Enable the feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/).
* Purchase a product with card `4000000000000259` to get disputed right away.
* Open the transaction details page of that new order by going to admin > Payments > Transactions > click on the top most transaction.
* Confirm there is 'Steps to resolve' section.
* Confirm the steps are:
```
1. Email the customer to address their concerns.
2. Provide guidance on dispute withdrawal if the customer agrees.
3. Challenge (?) or accept (?) the dispute by {dispute due by date} ({n} days left to respond)
```
* Click on the 'Email the customer' link. It should open your local email client. Confirm:
  * To address is the customer's email address.
  * Subject is `Problem with your purchase from {store name} on {YYYY-MM-DD of the charge date}?`
  * Body is:
```
Hello {customer name}

We noticed that on {YYYY-MM-DD of the dispute created date}, you disputed a {currency formatted dispute amount, eg: $10.00} from {YYYY-MM-DD of the charge date}. We wanted to contact you to make sure everything was all right with your purchase and see if there's anything else we can do to resolve any problems you might have had.

Alternatively, if the dispute was a mistake, you could easily withdraw it by calling the number on the back of your card. Thank you so much - we appreciate your business and look forward to working with you.
```
* Confirm the 2nd step has 'guidance on dispute withdrawal' link to https://woocommerce.com/document/woopayments/fraud-and-disputes/managing-disputes/#withdrawals.
* Confirm the 3rd step has icons:
  * The first icon when clicked will explain about challenging the dispute:
```
Challenge the dispute if you consider the claim invalid. You'll need to provide evidence to back your claim. Keep in mind that challenging doesn't ensure a resolution in your favor.
```
  * The second icon when clicked will explain about accepting the dispute:
```
Accepting this dispute will automatically close it. Your account will be charged a {currency formatted dispute fee, eg: $15.00} fee, and the disputed amount will be refunded to the cardholder.
```
* Confirm the 3rd step shows the dispute's due by date:
  * You can either edit [this code](https://github.com/Automattic/woocommerce-payments/blob/04ee3f67f48e6bb5dcf3deb34986fc89f879ebf8/client/payment-details/dispute-details/dispute-steps.tsx#L156-L160) or pass `daysRemaining` at your will [here](https://github.com/Automattic/woocommerce-payments/blob/04ee3f67f48e6bb5dcf3deb34986fc89f879ebf8/client/payment-details/dispute-details/index.tsx#L70), or hardcode the dispute's due by date to test these cases:
  * When due date is still 7-day away or more, the text color is black.
		Screenshot 2023-09-14 at 06 59 20
  * When due date is between 2 and 7-day away, the text color is yellow.
		Screenshot 2023-09-14 at 07 02 53
  * When due date is less than 3-day away, the text color is red.
		Screenshot 2023-09-14 at 07 04 36
  * When due date is today, the text color is red and it says `Last day today`.
		Screenshot 2023-09-14 at 07 06 45

**Inquiries**
* Purchase a product with card `4000000000001976` to create an inquiry.
* Open the transaction details page of that new order by going to admin > Payments > Transactions > click on the top most transaction.
* Confirm that the card is rendered correctly with the appropriate styling, but no "Steps to Resolve" section is displayed.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7206` or branch name `add/6925-steps-to-resolve` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/6925-steps-to-resolve) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 71a334945fdfcaf6ae5a3415174b87fbf86625ec
- Build time: 2023-09-21 22:32:23 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx]: I intentionally do not translate the email subject and body because it is a bit weird to translate a content that is for the customer who might not speak the same language as merchant.

What do you think @souravdebnath1986 @nikkivias? Should we send this email as merchant's language or in English? Or maybe both as I've seen that practice (sending in 2 languages in one email) a lot, but if so, what about the subject?

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx:70]: Wanted to use [querystring.strinfigy()](https://nodejs.org/api/querystring.html#querystringstringifyobj-sep-eq-options) but I don't see it's used at all in the codebase, so I just use `encodeURIComponent()` to escape each param's value.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx:165]: I copied this from [DisputeSummaryRow's](https://github.com/Automattic/woocommerce-payments/blob/a23c791887e8776ba1f19ffdbbb337697346439f/client/payment-details/dispute-details/dispute-summary-row.tsx#L96-L119) so it looks consistent.

One doubt I have is that in the design paJDYF-9Ip-p2, when due date is still more than 7-day away, it does not show the `({n} days left to respond)` part.

Screenshot 2023-09-14 at 07 16 13

but I see in `DisputeSummaryRow`, it shows that text, so for consistency sake, I follow `DisputeSummaryRow`.

File: client/payment-details/dispute-details/style.scss
Comment (shendy-a8c) [inline client/payment-details/dispute-details/style.scss]: This is necessary to fix the line break problem shown here:

https://github.com/Automattic/woocommerce-payments/assets/73803630/8201edfe-ca4e-4c77-a37d-dc9312a07a8c

File: client/payment-details/dispute-details/style.scss
Comment (shendy-a8c) [inline client/payment-details/dispute-details/style.scss]: This is so the icon is positioned just right.

https://github.com/Automattic/woocommerce-payments/assets/73803630/497651c6-c437-4936-8383-8f1621d76aa2

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx]: Waiting on https://github.com/Automattic/woocommerce-payments/pull/7118.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx]: This will be resolved later in https://github.com/Automattic/woocommerce-payments/issues/7156.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx]: Asked p1694651166956509-slack-C03LA109BKM.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [inline client/payment-details/dispute-details/dispute-steps.tsx]: Need to check when customer name is null, will it say undefined or just empty string.

**Comment by shendy-a8c** [comment]
Just to note that I'm open for suggestion to improve this PR, especially because of prop-drilling, ie since `DisputeSteps` needs data outside the dispute object like customer name, email, charge created, etc. Maybe that cannot be avoided? Or maybe pass the whole charge object instead of just dispute (or maybe not)?

**Comment by shendy-a8c** [comment]
Even though there are couple blockers, I think this PR can be reviewed already.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [reply client/payment-details/dispute-details/dispute-steps.tsx]: Addressed in a7ddf9485db740de0da816e11f9a3f185a8dfbfc.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (shendy-a8c) [reply client/payment-details/dispute-details/dispute-steps.tsx]: When customer's name is null, it'd be printed literarily as 'null' so addressed in 116a8db1b3bde6ac8062c2b86cd0486f949564f6.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (haszari) [reply client/payment-details/dispute-details/dispute-steps.tsx]: Can we keep it simple? This is site content, so it should be translatable. Just like content on pages, buttons, or in site emails. 

The merchant has control of all of this. The default experience should not have mixed languages IMO.

Comment (haszari) [reply client/payment-details/dispute-details/dispute-steps.tsx]: Note – this will not _send_ an email, just generate a draft email in merchant's email client.

**Comment by Jinksi** [comment]
I've removed the **Review the claim** step from this PR's scope in e8a83fe – now handled in issue #7156.

**Comment by Jinksi** [comment]
I've removed the `blocked` status since:
* **Review the claim** is not part of this issue – now handled in issue #7156.
* The draft [guidance on dispute withdrawal](https://woocommerce.com/document/woopayments/fraud-and-disputes/managing-disputes/#withdrawals) documentation page has been published.

File: client/payment-details/dispute-details/style.scss
Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: Nice catch! I've moved this to the tooltip component styles in 0b768b3

File: client/payment-details/dispute-details/style.scss
Comment (Jinksi) [reply client/payment-details/dispute-details/style.scss]: I've also moved the `vertical-align: text-bottom` as a default tooltip style in 6a55603

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-steps.tsx]: Resolved in 3d35ecc

**Comment by Jinksi** [comment]
FYI @shendy-a8c I've resolved a few of the TODOs and discussion points for this PR.

~From what I see, it's just the email translation discussion to resolve~: https://github.com/Automattic/woocommerce-payments/pull/7206#discussion_r1330840913.

**Update:** I've made the email subject and body translatable in 787b8d9 to let the merchant decide what language to use.

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (haszari) [reply client/payment-details/dispute-details/dispute-steps.tsx]: For example, if a multilingual merchant wants to send a localised "is everything ok with your purchase" email, they can edit in their mail client of choice or use another mailing solution such as MailPoet etc.

Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-steps.tsx]: I've made the email subject and body translatable in 787b8d9 to let the merchant decide what language to use.

**Comment by Jinksi** [comment]
This PR is ready for another review. FYI @shendy-a8c 🙏

**Review by shendy-a8c — COMMENTED** [review]
Looks great, @Jinksi !

Tested:
- With dispute, I could see the steps and actions.
- With inquiry, I did not see the steps and actions.
- The tooltip CSS move made me nervous because that means it will affect all `TooltipBase` usage, ie `ClickTooltip` and `HoverTooltip` but I checked they all look good as if that CSS does not do anything =p. Checking `HoverTooltip` is a bit much but mainly the settings page and the payment logo (eg Visa's) on the transactions list page. I see it's used in WooPay as well but I don't know how to test that.

Thank you for the great work!

I can't approve this PR as I authored it, so go ahead approve and merge.

File: client/components/tooltip/style.scss
Comment (haszari) [inline client/components/tooltip/style.scss]: Minor nitpick – should our CSS comments be sentences with full stop? I thought we had a linter enforcing this like PHP / JS. Not a blocker just curious what our standard is.

Side note (also nitpick, not a blocker) – I find it clearer when css comments state the positive outcome/purpose of the rule, rather than define in terms of a bad behaviour that's fixed. Not sure what that would look like for this one.

File: client/payment-details/dispute-details/dispute-awaiting-response-details.tsx
Comment (haszari) [inline client/payment-details/dispute-details/dispute-awaiting-response-details.tsx:61]: Just noting this is temporary – a comment might be helpful here (i.e. this should get deleted when inquiry version is done).

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (haszari) [reply client/payment-details/dispute-details/dispute-steps.tsx:70]: Has anyone tested clicking the link correctly works with at least one email client? If there's any issue with escaping or protocol then that would catch it! I have this behaviour disabled on my mac (because I use GMail), so wasn't able to test. 

Noting so we can test in our test swarm today e.g. on mobile or if someone has `mailto` configured right :)

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (haszari) [inline client/payment-details/dispute-details/dispute-steps.tsx:127]: I found these inline icon tooltips a bit weird, they break the flow of the sentence. It's functioning well though and super clear and usable so not a blocker. 

Maybe that info could be in a single help popup, nearer the action buttons. Thinking out loud – @nikkivias  is our UX expert here.

**Review by haszari — APPROVED** [review]
Did a quick review and tested one dispute – looking good and working well. Based on previous reviews this is good to merge so I will merge! 🚢

File: client/components/tooltip/style.scss
Comment (Jinksi) [reply client/components/tooltip/style.scss]: Yeh, the linter enforces this for PHP.

Suggestion with the positive outcome/purpose:
```suggestion
		// Ensure that the tooltip doesn't shift neighboring elements when it appears.
```

File: client/payment-details/dispute-details/dispute-awaiting-response-details.tsx
Comment (Jinksi) [reply client/payment-details/dispute-details/dispute-awaiting-response-details.tsx:61]:
```suggestion
	// This is a temporary restriction and can be removed once steps and actions for inquiries are implemented.
	const showDisputeStepsAndActions = ! isInquiry( dispute );
```

**Comment by Jinksi** [comment]
> Has anyone tested clicking the link correctly works with at least one email client? If there's any issue with escaping or protocol then that would catch it! 

@haszari I have tested the mailto links with macOS mail, works well!

I'll share a screenshot when I get the chance.

**Comment by shendy-a8c** [comment]
> Has anyone tested clicking the link correctly works with at least one email client? If there's any issue with escaping

I have tested it and it worked but in English. If we really want to be sure escaping works, it should be tested with non-ASCII charset languages like Chinese, Japanese, Arabic, etc. I will try it during testing. Good call out!

Update: hmm but since the strings won't be translated yet, I think it has to be tested manually.

**Comment by Jinksi** [comment]
Disabling auto-merge, will merge this manually once GH checks complete.

**Comment by Jinksi** [comment]
> I'll share a screenshot when I get the chance.

An example email in macOS that appears when clicking `Email the customer`. 

image

cc @haszari

File: client/payment-details/dispute-details/dispute-steps.tsx
Comment (dominique-vias) [reply client/payment-details/dispute-details/dispute-steps.tsx]: I think to default to the merchant's language is better than to suddenly see English if they have localized configuration.

**Pull request merged**
